src/flux/math.py

```diff
@@ -38,16 +37,32 @@ def attention(q: Tensor, k: Tensor, v: Tensor, pe: Tensor) -> Tensor:
     q, k = apply_rope(q, k, pe)
 
     if xformers_flash3:
+        if torch_sdpa or triton_attention:
+            print(
+                "Warning: xformers_flash3 is enabled, but torch_sdpa or triton_attention is also enabled. "
+                "Please remain only one of them."
+            )
+
         q = q.permute(0, 2, 1, 3) # B, H, S, D
         k = k.permute(0, 2, 1, 3) # B, H, S, D
         v = v.permute(0, 2, 1, 3) # B, H, S, D
         
-        x = compiled_xformers_flash_hopper(q, k, v).permute(0,2,1,3)
-    if torch_sdpa:
+        x = _compiled_xformers_flash_hopper(q, k, v).permute(0,2,1,3)
+    elif torch_sdpa:
+        if triton_attention:
+            print(
+                "Warning: torch_sdpa is enabled, but triton_attention is also enabled. "
+                "Please remain only one of them."
+            )
+
         x = scaled_dot_product_attention(q, k, v)
-    if triton_attention:
+    elif triton_attention:
+        from triton.ops import attention as attention_triton
+
         softmax_scale = q.size(-1) ** -0.5
         x = attention_triton(q, k, v, True, softmax_scale)
+    else:
+        x = torch.nn.functional.scaled_dot_product_attention(q, k, v)
```

Comment:
Refactored attention mechanism implementation to improve code maintainability and provide clearer error handling when multiple attention methods are enabled simultaneously. Major changes include:
* Lazy importing of optional dependencies (xformers, triton)
* Added warning messages for conflicting attention method selections
* Made default attention method explicit (torch.scaled_dot_product_attention)